src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1191 @@
+            { InternationalFixedDate.of(2011,  5, 26), InternationalFixedDate.of(2111,  5, 25), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2011,  5, 26), InternationalFixedDate.of(2111,  5, 26), ChronoUnit.CENTURIES, 1 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(2014,  5, 26), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(2114,  5, 25), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(2114,  5, 26), ChronoUnit.CENTURIES, 1 },
+            { InternationalFixedDate.of(2012, true, false), InternationalFixedDate.of(2012, true, false), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2012, true, false), InternationalFixedDate.of(2108, true, false), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2012, true, false), InternationalFixedDate.of(2312, 10,  7), ChronoUnit.CENTURIES, 3 },
+            { InternationalFixedDate.of(2012, true, false), InternationalFixedDate.of(1639,  2, 19), ChronoUnit.CENTURIES, -3 },
+            { InternationalFixedDate.of(2012, true, false), InternationalFixedDate.of(2678, false, true), ChronoUnit.CENTURIES, 6 },
+            { InternationalFixedDate.of(2012, false, true), InternationalFixedDate.of(2012, false, true), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2012, false, true), InternationalFixedDate.of(2022, 13, 28), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2012, false, true), InternationalFixedDate.of(2238, false, true), ChronoUnit.CENTURIES, 2 },
+            { InternationalFixedDate.of(2012, false, true), InternationalFixedDate.of(1124,  4, 17), ChronoUnit.CENTURIES, -8 },
+            { InternationalFixedDate.of(2011, false, true), InternationalFixedDate.of(2011, false, true), ChronoUnit.CENTURIES, 0 },
+            { InternationalFixedDate.of(2011, false, true), InternationalFixedDate.of(2345,  3, 23), ChronoUnit.CENTURIES, 3 },
+            { InternationalFixedDate.of(2011, false, true), InternationalFixedDate.of(1787,  8,  6), ChronoUnit.CENTURIES, -2 },
+
+            { InternationalFixedDate.of(2011,  5, 26), InternationalFixedDate.of(2011,  5, 26), ChronoUnit.MILLENNIA, 0 },
+            { InternationalFixedDate.of(2011,  5, 26), InternationalFixedDate.of(3011,  5, 25), ChronoUnit.MILLENNIA, 0 },
+            { InternationalFixedDate.of(2011,  5, 26), InternationalFixedDate.of(5011,  5, 26), ChronoUnit.MILLENNIA, 3 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(2014,  5, 26), ChronoUnit.MILLENNIA, 0 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(3014,  5, 25), ChronoUnit.MILLENNIA, 0 },
+            { InternationalFixedDate.of(2014,  5, 26), InternationalFixedDate.of(5014,  5, 26), ChronoUnit.MILLENNIA, 3 },
+       
```

Comment:
...why are you bothering with this particular guard clause?  The 28th day will _always_ be a Saturday, leap year or not.